mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -42,14 +42,23 @@ import org.apache.spark.util.VersionUtils.majorMinorVersion
  * otherwise the features will not be mapped evenly to the columns.
  */
 @Since("1.2.0")
-class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
+class HashingTF @Since("3.0.0") (
+    @Since("1.4.0") override val uid: String,
+    private[ml] val hashFuncVersion: Int)
   extends Transformer with HasInputCol with HasOutputCol with HasNumFeatures
     with DefaultParamsWritable {
 
-  private var hashFunc: Any => Int = FeatureHasher.murmur3Hash
+  @transient lazy val hashFunc = hashFuncVersion match {
```

Comment:
This is just an int parameter; can it not be a potentially private parameter like other things? just seemed odd to put it in the constructor when other params are not.

This doesn't need to be lazy or transient I think.

Reply:
@srowen  I think it'd better to be transient to ensure the transformer supporting serializing. It extends `Serializable` interface.

Reply:
I don't feel strongly about it either way, but it already serialized as a function before.